### PR TITLE
docker_swarm_service: Add option rollback_config

### PR DESCRIPTION
**Pull request by hannseman**
##### SUMMARY

This PR adds the option `rollback_config`. It provides an alternative to https://github.com/ansible/ansible/pull/51828 which author have not been responsive and the 2.8 code freeze is approaching.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

docker_swarm_service

### CODE REVIEW

File: lib/ansible/modules/cloud/docker/docker_swarm_service.py [hunk 1]
```diff
@@ -1575,6 +1655,8 @@ def compare(self, os):
             differences.add('restart_policy_delay', parameter=self.restart_policy_delay, active=os.restart_policy_delay)
         if self.restart_policy_window is not None and self.restart_policy_window != os.restart_policy_window:
             differences.add('restart_policy_window', parameter=self.restart_policy_window, active=os.restart_policy_window)
+        if self.has_rollback_config_changed(os.rollback_config):
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/docker/docker_swarm_service.py]:
```suggestion
        if self.rollback_config is not None and self.has_rollback_config_changed(os.rollback_config):
```

File: lib/ansible/modules/cloud/docker/docker_swarm_service.py [hunk 2]
```diff
@@ -1648,6 +1730,16 @@ def has_image_changed(self, old_image):
             old_image = old_image.split('@')[0]
         return self.image != old_image, old_image
 
+    def has_rollback_config_changed(self, old_rollback_config):
+        defined_options = dict(
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/docker/docker_swarm_service.py:1749]: You should abort early with `True` if `old_rollback_config is None and self.rollback_config`.

File: lib/ansible/modules/cloud/docker/docker_swarm_service.py (hunk 1 above)
Comment (hannseman) [reply lib/ansible/modules/cloud/docker/docker_swarm_service.py]: Good catch, actually I missed to default `self.rollback_config` as None if it was not defined. Fixed now.

**Comment by hannseman** [comment]
Appears like I accidentally commited some test-meta-data file which added the `core_review` and `support:core`-labesl. I've rebased away the file, hopefully ansibot will remove them automatically?

**Review by felixfontein — APPROVED** [review]
LGTM

**Review by jwitko — APPROVED** [review]
LGTM

**Comment by felixfontein** [comment]
@hannseman thanks for this PR!
@jwitko thanks for reviewing!
@willthames thanks for merging!

**Pull request merged**